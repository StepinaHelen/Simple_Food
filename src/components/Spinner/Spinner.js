import styled from "styled-components";

import { Hearts } from "react-loader-spinner";

export const Wrapper = styled.div`
  margin: 0 auto;
  text-align: center;
`;

const Spinner = () => {
  return (
    <Wrapper>
      <Hearts
        height="200"
        width="200"
        color="#6b068a"
        ariaLabel="hearts-loading"
        wrapperStyle={{ display: "block" }}
        wrapperClass=""
        visible={true}
      />
    </Wrapper>
  );
};
export default Spinner;
