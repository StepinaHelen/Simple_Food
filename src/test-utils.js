import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
