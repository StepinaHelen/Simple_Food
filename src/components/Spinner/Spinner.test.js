import { render } from "../../test-utils";
import Spinner from "./Spinner";
describe("test Spinner", () => {
  it("Spinner renders", () => {
    render(<Spinner />);
  });
  it("Spinner snapshot", () => {
    const view = render(<Spinner />);
    expect(view).toMatchSnapshot();
  });
});
