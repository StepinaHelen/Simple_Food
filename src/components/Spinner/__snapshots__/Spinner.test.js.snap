// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Spinner Spinner snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  margin: 0 auto;
  text-align: center;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          aria-busy="true"
          aria-label="hearts-loading"
          class=""
          data-testid="hearts-loading"
          role="status"
          style="display: block;"
        >
          <svg
            data-testid="hearts-svg"
            fill="#6b068a"
            height="200"
            viewBox="0 0 140 64"
            width="200"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              attributeName="fill-opacity"
              d="M30.262 57.02L7.195 40.723c-5.84-3.976-7.56-12.06-3.842-18.063 3.715-6 11.467-7.65 17.306-3.68l4.52 3.76 2.6-5.274c3.717-6.002 11.47-7.65 17.305-3.68 5.84 3.97 7.56 12.054 3.842 18.062L34.49 56.118c-.897 1.512-2.793 1.915-4.228.9z"
              from="0"
              to=".5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.4s"
                repeatCount="indefinite"
                values="0.5;1;0.5"
              />
            </path>
            <path
              attributeName="fill-opacity"
              d="M105.512 56.12l-14.44-24.272c-3.716-6.008-1.996-14.093 3.843-18.062 5.835-3.97 13.588-2.322 17.306 3.68l2.6 5.274 4.52-3.76c5.84-3.97 13.592-2.32 17.307 3.68 3.718 6.003 1.998 14.088-3.842 18.064L109.74 57.02c-1.434 1.014-3.33.61-4.228-.9z"
              from="0"
              to=".5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0.7s"
                calcMode="linear"
                dur="1.4s"
                repeatCount="indefinite"
                values="0.5;1;0.5"
              />
            </path>
            <path
              d="M67.408 57.834l-23.01-24.98c-5.864-6.15-5.864-16.108 0-22.248 5.86-6.14 15.37-6.14 21.234 0L70 16.168l4.368-5.562c5.863-6.14 15.375-6.14 21.235 0 5.863 6.14 5.863 16.098 0 22.247l-23.007 24.98c-1.43 1.556-3.757 1.556-5.188 0z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin: 0 auto;
  text-align: center;
}

<div>
    <div
      class="c0"
    >
      <div
        aria-busy="true"
        aria-label="hearts-loading"
        class=""
        data-testid="hearts-loading"
        role="status"
        style="display: block;"
      >
        <svg
          data-testid="hearts-svg"
          fill="#6b068a"
          height="200"
          viewBox="0 0 140 64"
          width="200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            attributeName="fill-opacity"
            d="M30.262 57.02L7.195 40.723c-5.84-3.976-7.56-12.06-3.842-18.063 3.715-6 11.467-7.65 17.306-3.68l4.52 3.76 2.6-5.274c3.717-6.002 11.47-7.65 17.305-3.68 5.84 3.97 7.56 12.054 3.842 18.062L34.49 56.118c-.897 1.512-2.793 1.915-4.228.9z"
            from="0"
            to=".5"
          >
            <animate
              attributeName="fill-opacity"
              begin="0s"
              calcMode="linear"
              dur="1.4s"
              repeatCount="indefinite"
              values="0.5;1;0.5"
            />
          </path>
          <path
            attributeName="fill-opacity"
            d="M105.512 56.12l-14.44-24.272c-3.716-6.008-1.996-14.093 3.843-18.062 5.835-3.97 13.588-2.322 17.306 3.68l2.6 5.274 4.52-3.76c5.84-3.97 13.592-2.32 17.307 3.68 3.718 6.003 1.998 14.088-3.842 18.064L109.74 57.02c-1.434 1.014-3.33.61-4.228-.9z"
            from="0"
            to=".5"
          >
            <animate
              attributeName="fill-opacity"
              begin="0.7s"
              calcMode="linear"
              dur="1.4s"
              repeatCount="indefinite"
              values="0.5;1;0.5"
            />
          </path>
          <path
            d="M67.408 57.834l-23.01-24.98c-5.864-6.15-5.864-16.108 0-22.248 5.86-6.14 15.37-6.14 21.234 0L70 16.168l4.368-5.562c5.863-6.14 15.375-6.14 21.235 0 5.863 6.14 5.863 16.098 0 22.247l-23.007 24.98c-1.43 1.556-3.757 1.556-5.188 0z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
